internal/replicated_map.go

```diff
@@ -0,0 +1,256 @@
+// Copyright (c) 2008-2018, Hazelcast, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License")
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package internal
+
+import (
+	. "github.com/hazelcast/hazelcast-go-client/core"
+	. "github.com/hazelcast/hazelcast-go-client/internal/common"
+	. "github.com/hazelcast/hazelcast-go-client/internal/protocol"
+	"github.com/hazelcast/hazelcast-go-client/internal/serialization"
+	. "github.com/hazelcast/hazelcast-go-client/serialization"
+	"math/rand"
+	"time"
+)
+
+type ReplicatedMapProxy struct {
+	*proxy
+	targetPartitionId int32
+}
+
+func newReplicatedMapProxy(client *HazelcastClient, serviceName *string, name *string) *ReplicatedMapProxy {
+	partitionCount := client.PartitionService.PartitionCount()
+	targetPartitionId := rand.Int31n(partitionCount)
+	return &ReplicatedMapProxy{proxy: &proxy{client, serviceName, name}, targetPartitionId: targetPartitionId}
+}
+func (rmp *ReplicatedMapProxy) Put(key interface{}, value interface{}) (oldValue interface{}, err error) {
+	keyData, valueData, err := rmp.validateAndSerialize2(key, value)
+	if err != nil {
+		return nil, err
+	}
+	request := ReplicatedMapPutEncodeRequest(rmp.name, keyData, valueData, TTL)
```

Comment:
`TTL` here comes from map.go. This is not good practice. I dont think this constant adds value. Straight `0` would be more readable. If you really want to use a constant, name it better (`TTL_UNLIMITED`) and make it private to this proxy.